From 7c5825c857fbb4e5a3a47b3edf605f871090b58f Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:07:45 -0400
Subject: [usb] option: add Alcatel L800MA

Message-id: <1410887463-20674-394-git-send-email-dzickus@redhat.com>
Patchwork-id: 95614
O-Subject: [RHEL7 PATCH 393/591] usb: option: add Alcatel L800MA
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit dd6b48ecec2ea7d15f28d5e5474388681899a5e1
Author: Bjorn Mork <bjorn@mork.no>
Date:   Fri Apr 25 18:49:19 2014 +0200

    usb: option: add Alcatel L800MA

    Device interface layout:
    0: ff/ff/ff - serial
    1: ff/00/00 - serial AT+PPP
    2: ff/ff/ff - QMI/wwan
    3: 08/06/50 - storage

    Cc: <stable@vger.kernel.org>
    Signed-off-by: Bjorn Mork <bjorn@mork.no>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/option.c b/drivers/usb/serial/option.c
index 6335222..776c86f 100644
--- a/drivers/usb/serial/option.c
+++ b/drivers/usb/serial/option.c
@@ -287,6 +287,7 @@ static void option_instat_callback(struct urb *urb);
 #define ALCATEL_PRODUCT_X060S_X200  0x0000
 #define ALCATEL_PRODUCT_X220_X500D  0x0017
 #define ALCATEL_PRODUCT_L100V   0x011e
+#define ALCATEL_PRODUCT_L800MA   0x0203
 
 #define PIRELLI_VENDOR_ID   0x1266
 #define PIRELLI_PRODUCT_C100_1   0x1002
@@ -1501,6 +1502,8 @@ static const struct usb_device_id option_ids[] = {
    .driver_info = (kernel_ulong_t)&net_intf5_blacklist },
  { USB_DEVICE(ALCATEL_VENDOR_ID, ALCATEL_PRODUCT_L100V),
    .driver_info = (kernel_ulong_t)&net_intf4_blacklist },
+ { USB_DEVICE(ALCATEL_VENDOR_ID, ALCATEL_PRODUCT_L800MA),
+   .driver_info = (kernel_ulong_t)&net_intf2_blacklist },
  { USB_DEVICE(AIRPLUS_VENDOR_ID, AIRPLUS_PRODUCT_MCD650) },
  { USB_DEVICE(TLAYTECH_VENDOR_ID, TLAYTECH_PRODUCT_TEU800) },
  { USB_DEVICE(LONGCHEER_VENDOR_ID, FOUR_G_SYSTEMS_PRODUCT_W14),
-- 
1.7.1